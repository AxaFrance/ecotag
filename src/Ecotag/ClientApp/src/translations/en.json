{
  "header": {
    "title": "An eco-system that aims to be eco-friendly",
    "errors": {
      "not_connected": "Not connected"
    }
  },
  "paging": {
    "display": "Display",
    "elements": "elements",
    "previous": "« Previous",
    "next": "Next »"
  },
  "stepper": {
    "confirm": "Confirm"
  },
  "resilience": {
    "error": {
      "first_part": "An error occurred.",
      "second_part": "We are aware of the problem. An investigation is ongoing..."
    },
    "forbidden": "You are not allowed to access this resource."
  },
  "not_found": {
    "title": "404 Page Not Found",
    "not": "not",
    "found": "found",
    "message": "The page you are looking for is not here !"
  },
  "home": {
    "title": "Home",
    "projects": "Projects",
    "datasets": "Datasets",
    "groups": "Teams",
    "errors": {
      "no_roles": "No role is associated to your profile."
    }
  },
  "group": {
    "title": "Teams",
    "subtitle": "Annotation teams governance",
    "creation_title": "Creation",
    "new": {
      "name_input": {
        "label": "Name : ",
        "help_message": "Ex : bank account details annotators"
      },
      "validation": "Validate"
    },
    "count_teams": {
      "first_part": "There is/are (",
      "second_part": ") team(s)"
    },
    "empty_list": "No element",
    "list": {
      "header": {
        "name": "Name",
        "users": "Users",
        "actions": "Actions"
      },
      "edit": "Edit",
      "edit_popup": {
        "header": "Add/Remove users from this team",
        "add_user_placeholder": "Add user",
        "footer": {
          "cancel": "Cancel",
          "validation": "Validate"
        }
      }
    }
  },
  "dataset": {
    "title": "Datasets",
    "subtitle": "A dataset is the representation of a collection of data",
    "new_dataset": "New dataset",
    "access_token": "Access token",
    "active_datasets_counter": "Your active datasets : ",
    "filter_by": "Filter by",
    "name": "Name",
    "empty_list": "No elements",
    "list": {
      "actions_titles": {
        "locked": "locked",
        "locked_and_work_in_progress": "locked and work in progress",
        "pending": "pending",
        "none": "unlocked"
      },
      "header": {
        "lock": "Lock",
        "name": "Name",
        "team": "Team",
        "classification": "Classification",
        "nb_files": "Number of files",
        "type": "Type",
        "creation_date": "Creation date",
        "action": "Action"
      },
      "edit": "Edit"
    },
    "new": {
      "title": "New dataset",
      "stepper": {
        "title": "New dataset"
      },
      "form_title": "New dataset",
      "name_input": {
        "label": "Name",
        "help_message": "Ex : bank-account-details-dataset"
      },
      "team_input": {
        "label": "Team"
      },
      "type_input": {
        "label": "Type",
        "help_message": "Ex : Image, Text",
        "options_labels": {
          "image": "Images (.jpg, .png, jpeg)",
          "document": "Documents (.txt, .eml, .tiff, .pdf, .doc, etc.)",
          "text": "Text (.txt)",
          "eml": "Emails (.eml)"
        },
        "help": {
          "title": "Information on the accepted documents types",
          "images": {
            "name": "Images",
            "types": " : .jpg, .png, jpeg"
          },
          "documents": {
            "name": "Documents",
            "types": " : .txt, .eml, .msg, .jpg, .png, jpeg, .tiff, .tif, .pdf, .doc, .docx, .xls, .xlsx, .ppt, .pptx, .rtf, .odt, .ods, .odp, .zip"
          },
          "text": {
            "name": "Text",
            "types": " : .txt"
          },
          "eml": {
            "name": "Eml",
            "types": " : .eml"
          }
        }
      },
      "classification_input": {
        "label": "Classification",
        "help_message": "Ex : Critical",
        "options": {
          "public": {
            "label": "Public"
          },
          "internal": {
            "label": "Internal"
          },
          "confidential": {
            "label": "Confidential"
          },
          "critical": {
            "label": "Critical"
          }
        },
        "help": {
          "title": "Information on criticality",
          "public": {
            "name": "Public",
            "detail": " : a piece of information made deliberately for the public, or specifically made for public diffusion"
          },
          "internal": {
            "name": "Internal",
            "detail": " : a piece of information that is accessible to all or a large portion of employees and is not intended for external parties."
          },
          "confidential": {
            "name": "Confidential",
            "detail": " : a piece of information restricted to a specific audience"
          },
          "critical": {
            "name": "Critical",
            "detail": " : Please refer to your manager."
          }
        }
      },
      "import_input": {
        "label": "Import from a blob : "
      },
      "dataset_to_import": {
        "label": "Dataset to import : "
      },
      "validate": "Validation"
    },
    "confirm": {
      "title": "Confirmation",
      "stepper": "New dataset",
      "confirmation_title": "New dataset added !",
      "confirmation_description": "If you have imported a dataset, the process of adding files can take a while.",
      "return_button_label": "Go back to datasets list"
    },
    "edit": {
      "confirm_modal": {
        "title": "Do you want to lock the dataset permanently ?",
        "warning_first_part": "This action is permanent.",
        "warning_second_part": "Every modification (addition or deletion of files) will be impossible then."
      },
      "subtitle": "Dataset edition",
      "lock": "Lock",
      "locked_state": {
        "locked": "Dataset locked",
        "locked_and_work_in_progress": "Dataset is being worked on",
        "default": "Dataset is being imported"
      },
      "file_upload": {
        "title": "Upload files ",
        "file_input": {
          "label": "Browse",
          "placeholder": "Drag/drop your files here"
        },
        "send_button": {
          "label": "Send"
        },
        "errors": {
          "already_existing": {
            "first_part": "The file ",
            "second_part": " already exists"
          },
          "could_not_upload": {
            "first_part": "The file ",
            "second_part": " could not be uploaded"
          }
        }
      },
      "files_list": {
        "global_info_tab": {
          "title": "Global information",
          "name": "Name",
          "creation_date": "Creation date",
          "type": "Type",
          "nb_files": "Number of files",
          "files_total_size": "Total files size",
          "files_mean_size": "Average file size"
        },
        "files_list_tab": {
          "title": "Files list",
          "name": "Name",
          "type": "Type",
          "size": "Size",
          "action": "Action",
          "delete_action": "Delete"
        }
      }
    }
  }
}