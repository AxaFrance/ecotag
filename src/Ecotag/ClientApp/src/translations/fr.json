{
  "header": {
    "title": "Un écosystème qui vise à être écologique",
    "errors": {
      "not_connected": "Non connecté"
    }
  },
  "home": {
    "title": "Accueil",
    "projects": "Projets",
    "datasets": "Datasets",
    "groups": "Groupes",
    "errors": {
      "no_roles": "Aucun rôle n'est attribué à votre profil."
    }
  }
}