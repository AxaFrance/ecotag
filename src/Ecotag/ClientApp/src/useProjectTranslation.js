import './i18n';
import {useTranslation} from "react-i18next";
import i18next from 'i18next';

const useProjectTranslation = (namespace = 'translation') => {
    const {t: translate} = useTranslation(namespace);
    return {translate};
};

export const changeProjectTranslationLanguage = (language) => {
    i18next.changeLanguage(language);
};

export default useProjectTranslation;
