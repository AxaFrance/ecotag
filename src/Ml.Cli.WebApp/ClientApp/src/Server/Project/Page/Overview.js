import React from 'react';
import {
  ArticleRestitution,
  SectionRestitution,
  SectionRestitutionColumn,
  SectionRestitutionRow,
  Restitution,
  HeaderRestitution
} from '@axa-fr/react-toolkit-restitution';
import {formatTimestampToString} from "../../date";

export const Overview = ({ project, dataset, group }) => {
  //const labels = project.labels && project.labels.length > 0 ? project.labels : [];
  //const percentAdvancement = ((100 * (dataset.numberFiles - project.numberTagToDo)) / dataset.numberFiles).toFixed(2);

  return (
    <>
      <ArticleRestitution>
        <HeaderRestitution
            title="Informations générales"
        />
        <SectionRestitution>
          <SectionRestitutionRow >
            <SectionRestitutionColumn>
              <Restitution label="Nom" value={project.name} />
              <Restitution label="Type d'annotation" value={project.typeAnnotation} />
              <Restitution label="Date de création" value={formatTimestampToString(project.createDate)} />
              <Restitution label="Nombre de fichier" value={dataset.files.length} />
              <Restitution label="Nombre annotation croisée" value={project.numberCrossAnnotation} />
              
            </SectionRestitutionColumn>
            <SectionRestitutionColumn>
              <Restitution label="Dataset" value={dataset.name} />
              <Restitution label="Type de dataset" value={dataset.type} />
              <Restitution label="Groupe annotateurs" value={group.name} />
              <Restitution label="Pourcentage d&apos;avancement" value="100%" />
            </SectionRestitutionColumn>
          </SectionRestitutionRow>
        </SectionRestitution>
      </ArticleRestitution>    

    </>
  );
};

export default Overview;
