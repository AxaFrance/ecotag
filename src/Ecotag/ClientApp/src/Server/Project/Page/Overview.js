import React from 'react';
import {
    ArticleRestitution,
    HeaderRestitution,
    Restitution,
    SectionRestitution,
    SectionRestitutionColumn,
    SectionRestitutionRow
} from '@axa-fr/react-toolkit-restitution';
import {formatTimestampToString} from "../../date";
import useProjectTranslation from "../../../translations/useProjectTranslation";

export const Overview = ({project, dataset, group, annotationsStatus, users = []}) => {

    const {translate} = useProjectTranslation();

    const groupEmails = group.userIds.map(userId => {
        const user = users.find(user => user.id === userId);
        if (user == null) {
            return "";
        }
        return user.email;
    })

    return (
        <>
            <ArticleRestitution>
                <HeaderRestitution
                    title={translate('project.project_page.overview.title')}
                />
                <SectionRestitution>
                    <SectionRestitutionRow>
                        <SectionRestitutionColumn>
                            <Restitution label={translate('project.project_page.overview.name')} value={project.name}/>
                            <Restitution label={translate('project.project_page.overview.annotation_type')} value={project.annotationType}/>
                            <Restitution label={translate('project.project_page.overview.creation_date')} value={formatTimestampToString(project.createDate)}/>
                            <Restitution label={translate('project.project_page.overview.nb_files')} value={dataset.files.length}/>
                            <Restitution label={translate('project.project_page.overview.nb_cross_annotation')} value={project.numberCrossAnnotation}/>
                            <Restitution label={translate('project.project_page.overview.annotations_progress')}
                                         value={annotationsStatus ? `${annotationsStatus.percentageNumberAnnotationsDone}%` : '0%'}/>
                        </SectionRestitutionColumn>
                        <SectionRestitutionColumn>
                            <Restitution label={translate('project.project_page.overview.dataset')} value={dataset.name}/>
                            <Restitution label={translate('project.project_page.overview.dataset_type')} value={dataset.type}/>
                            <Restitution label={translate('project.project_page.overview.team')} value={group.name}/>
                            <Restitution label={translate('project.project_page.overview.emails')} values={groupEmails}/>
                        </SectionRestitutionColumn>
                    </SectionRestitutionRow>
                </SectionRestitution>
            </ArticleRestitution>
        </>
    );
};

export default Overview;
