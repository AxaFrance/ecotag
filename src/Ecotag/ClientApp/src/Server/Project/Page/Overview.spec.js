import '@testing-library/jest-dom/extend-expect';
import React from 'react';
import {render, waitFor} from '@testing-library/react';
import {Overview} from './Overview';
import {BrowserRouter as Router} from "react-router-dom";
import {changeProjectTranslationLanguage} from "../../../translations/useProjectTranslation";

const project = {
    "id": "0001",
    "name": "Information statement",
    "datasetId": "0004",
    "numberTagToDo": 10,
    "createDate": new Date("04-04-2011").getTime(),
    "annotationType": "NER",
    annotationStatus: {
        percentageNumberAnnotationsDone: 10
    }
};
const dataset = {
    "id": "0001",
    "name": "Green card",
    "type": "Image",
    "numberFiles": 300,
    "createDate": new Date("10-30-2019").getTime(),
    files: []
};
const group = {
    "id": "0001",
    "name": "developers",
    "userIds": ["0001", "0002"]
};
const users = [
    {
        id: "0001",
        "email": "guillaume.chervet@axa.fr",
        "nameIdentifier": "S000007"
    },
    {
        id: "0002",
        "email": "lilian.delouvy@axa.fr",
        "nameIdentifier": "S000005"
    }
];
const annotationStatus = {
    isAnnotationClosed: true,
    numberAnnotationsByUsers: [{
        "nameIdentifier": "S000005",
        numberAnnotations: 15
    }, {"nameIdentifier": "S000007", numberAnnotations: 35}],
    numberAnnotationsDone: 46,
    numberAnnotationsToDo: 288,
    percentageNumberAnnotationsDone: 32
};

describe('Overview', () => {
    const renderComponentAndCheckSnapshot = async(expectedText) => {
        const regex = new RegExp(expectedText, 'i');
        const {asFragment, getByText} = render(<Router><Overview dataset={dataset} project={project} group={group}
                                                                 annotationsStatus={annotationStatus}
                                                                 users={users}/></Router>);
        const messageEl = await waitFor(() => getByText(regex));
        expect(messageEl).toHaveTextContent(
            expectedText
        );
        expect(asFragment()).toMatchSnapshot();
    };
    it('should render correctly in english', async () => {
        changeProjectTranslationLanguage('en');
        await renderComponentAndCheckSnapshot("General information");
    });
    it('should render correctly in french', async () => {
        changeProjectTranslationLanguage('fr');
        await renderComponentAndCheckSnapshot("Informations générales");
    });
});

