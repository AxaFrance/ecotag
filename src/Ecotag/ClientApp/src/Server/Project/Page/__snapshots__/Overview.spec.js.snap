// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview should render correctly in english 1`] = `
<DocumentFragment>
  <article
    class="af-restitution"
  >
    <header
      class="af-restitution__header"
    >
      <div
        class="af-restitution__header-left"
      >
        <div
          class="af-restitution__title"
        >
          <span
            class="af-restitution__title-main"
          >
            General information
          </span>
          <span
            class="af-restitution__title-subtitle"
          />
        </div>
      </div>
    </header>
    <section
      class="af-restitution__content"
    >
      <div
        class="col col-sm-12 col-md-12 col-lg-12 col-xl-12"
      >
        <div
          class="row af-restitution__content-left"
        >
          <div
            class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
          >
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Name
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Information statement
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Annotation type
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                NER
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Creation date
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                02/01/0001
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Number of files
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                0
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Number of cross annotations
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                -
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Annotations progress
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                32%
              </dd>
            </dl>
          </div>
          <div
            class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
          >
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Dataset
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Green card
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Dataset type
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Image
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Annotators team
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                developers
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Team emails
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                <ul
                  class="af-restitution__listul"
                >
                  <li
                    class="af-restitution__listul-item"
                  >
                    guillaume.chervet@axa.fr
                  </li>
                  <li
                    class="af-restitution__listul-item"
                  >
                    lilian.delouvy@axa.fr
                  </li>
                </ul>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </section>
  </article>
</DocumentFragment>
`;

exports[`Overview should render correctly in french 1`] = `
<DocumentFragment>
  <article
    class="af-restitution"
  >
    <header
      class="af-restitution__header"
    >
      <div
        class="af-restitution__header-left"
      >
        <div
          class="af-restitution__title"
        >
          <span
            class="af-restitution__title-main"
          >
            Informations générales
          </span>
          <span
            class="af-restitution__title-subtitle"
          />
        </div>
      </div>
    </header>
    <section
      class="af-restitution__content"
    >
      <div
        class="col col-sm-12 col-md-12 col-lg-12 col-xl-12"
      >
        <div
          class="row af-restitution__content-left"
        >
          <div
            class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
          >
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Nom
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Information statement
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Type d'annotation
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                NER
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Date de création
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                02/01/0001
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Nombre de fichiers
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                0
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Nombre d'annotations croisées
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                -
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Avancement annotations
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                32%
              </dd>
            </dl>
          </div>
          <div
            class="col-sm-12 col-md-12 col-lg-6 col-xl-6"
          >
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Dataset
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Green card
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Type de dataset
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                Image
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Équipe d'annotateurs
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                developers
              </dd>
            </dl>
            <dl
              class="af-restitution__listdef"
            >
              <dt
                class="af-restitution__listdef-item"
              >
                <span
                  class="af-restitution__text"
                >
                  Emails de l'équipe
                </span>
              </dt>
              <dd
                class="af-restitution__listdef-value"
              >
                <ul
                  class="af-restitution__listul"
                >
                  <li
                    class="af-restitution__listul-item"
                  >
                    guillaume.chervet@axa.fr
                  </li>
                  <li
                    class="af-restitution__listul-item"
                  >
                    lilian.delouvy@axa.fr
                  </li>
                </ul>
              </dd>
            </dl>
          </div>
        </div>
      </div>
    </section>
  </article>
</DocumentFragment>
`;
